[OYPD-202] Theme updates for the location finder, adds templates, update styling…

david-hernandez opened the pull request:
- [x] Make sure your site has branch, camp, and facility content nodes.
- [x] Create landing page.
- [x] Add location finder filters paragraph to the header area.
- [x] Add location finder paragraph to the content area.
- [x] View page.

![screen shot 2017-02-14 at 10 18 36 pm](https://cloud.githubusercontent.com/assets/1504038/22959820/3325e4d0-f307-11e6-9191-5a9dac4ae741.png)

![screen shot 2017-02-14 at 10 21 27 pm](https://cloud.githubusercontent.com/assets/1504038/22959821/33297898-f307-11e6-88e7-fbfbc1dc76d7.png)

![screen shot 2017-02-14 at 10 21 34 pm](https://cloud.githubusercontent.com/assets/1504038/22959819/3324cece-f307-11e6-976e-6d0123ca9a09.png)

![screen shot 2017-02-14 at 10 25 20 pm](https://cloud.githubusercontent.com/assets/1504038/22959818/3324ef44-f307-11e6-868b-12be2350a192.png)

We may need to discuss the buttons. They are blue because of the color module implementation changes. I can add the btn-default styling to the colors.css file, but I'm concerned about any regressions that might cause.

david-hernandez committed:
Theme updates for the location finder, adds templates, update styling, update view

gundevel commented:
Build comment file: 
--------------
===============
<strong>2</strong>  errors for CodeSniffer: Drupal standard file http://openy-dev.ffwua.com/build469/build_reports/Drupalsniff.txt
Build site installed at http://openy-dev.ffwua.com/build469

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/469/

gundevel commented:
Behat test results file URL: http://openy-dev.ffwua.com/build469/profiles/openy/build/reports/behat

alexschedrov commented:
1. CodeSniffer notices
2. Bundle buttons should be have gray color
![image](https://cloud.githubusercontent.com/assets/1316234/22970467/f28b8096-f37a-11e6-98a9-fdc1f73cf1c6.png)

3. When "Coming soon" is selected, teaser should have following design + Branch title should lead to temporary URL.
![image](https://cloud.githubusercontent.com/assets/1316234/22970570/4b93ecbe-f37b-11e6-9227-5a7896d43e35.png)

Other than that looks awesome!

david-hernandez committed:
Updating for coming soon styling, fixing codesniffer error

gundevel commented:
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build473

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/473/

gundevel commented:
Behat test results file URL: http://openy-dev.ffwua.com/build473/profiles/openy/build/reports/behat

david-hernandez commented:
The last commit should handle the codesniffer error and the coming soon.

![screen shot 2017-02-15 at 9 23 33 am](https://cloud.githubusercontent.com/assets/1504038/22979865/d66edbfc-f365-11e6-8487-fae635d7d7fd.png)

david-hernandez committed:
Add temp url to branch teaser title, override for default white button color

david-hernandez commented:
Last commit adds white buttons and temp url. There is a problem where the buttons are going to get overridden if the main background color is changed. I haven't figure out how to separate them. They both start out with the same color.

Screenshot show I added a temp url that appears when I hovered over the title.

![screen shot 2017-02-15 at 5 03 13 pm](https://cloud.githubusercontent.com/assets/1504038/22998151/7d5485e2-f3a3-11e6-85f3-eab0e4ca7db9.png)

gundevel commented:
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build477

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/477/

gundevel commented:
Behat test results file URL: http://openy-dev.ffwua.com/build477/profiles/openy/build/reports/behat

Sanchiz commented:
retest this please

bjwilkins commented:
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build479

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/479/

alexschedrov commented on themes/openy_themes/openy_rose/templates/node/node--branch--teaser.html.twig:
```diff
@@ -81,13 +81,12 @@
   {% if not page %}
     <h2{{ title_attributes }}>
-      <a href="{{ url }}" rel="bookmark">{{ label }}</a>
+      <a href="{{ node.field_location_temp_url.0.url ? node.field_location_temp_url.0.url : url }}" rel="bookmark">{{ label }}</a>
```
1. This should be applied only when branch is "Coming Soon".
2. Let's set variable using `set` and the re-use it. `node.field_location_temp_url.0.url` looks a little bit scary.

bjwilkins commented:
Behat test results file URL: http://openy-dev.ffwua.com/build479/profiles/openy/build/reports/behat

alexschedrov commented on themes/openy_themes/openy_rose/templates/field/field--node--field-location-state.html.twig:
```diff
@@ -0,0 +1,69 @@
+  {% else %}
+    {% for item in items %}
+      {% if item.content['#markup']|length %}
+        <div{{ attributes.addClass(field_name) }}><i class="fa fa-star"></i><span>New Y</span>{{ item.content }}</div>
```
`New Y` should be translatable.

david-hernandez committed:
Adding translation, logic for temp url

bjwilkins commented:
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build482

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/482/

david-hernandez commented:
Added translation for coming soon text. Added logic to only use the temporary url when coming soon is activated for the node.

The `node.field_location_temp_url.0.url` is unfortunately needed. The url field does not supply the url separate from the text. There isn't a widget option for that. Also, the other cleaner options like `field_location_temp_url.uri` or `url` can return either an external url or a route. There is a url() function to deal with a route but not an external url. So every method I tried resulted in having to just dig into the url array.

bjwilkins commented:
Behat test results file URL: http://openy-dev.ffwua.com/build482/profiles/openy/build/reports/behat

AlexNetman commented:
Please, remove the scroll from pins
![image](https://cloud.githubusercontent.com/assets/4558228/23025603/8b9cda88-f46f-11e6-8918-b074085ae4e1.png)

alexschedrov commented:
![image](https://cloud.githubusercontent.com/assets/1316234/23025645/5440384e-f467-11e6-9eab-e365a0d63bd6.png)

AlexNetman commented:
![image](https://cloud.githubusercontent.com/assets/4558228/23025907/780e09aa-f470-11e6-8a0f-7b28686bc8ac.png)

Regarding "Coming soon" link -  I think it's better to just make redirect from node. But not updating all those view modes. Global redirect from location node to that URL would work for all other places were link to location might appear - menus, other landing pages, blogs associated with the branch.

david-hernandez committed:
Moving coming soon to top and floating so title will wrap

david-hernandez committed:
Removing the min height from the tooltip teasers for location finder

bjwilkins commented:
Build comment file: 
--------------
===============
Build site installed at http://openy-dev.ffwua.com/build484

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy-dev.ffwua.com:8080/job/PR_BUILDER/484/

bjwilkins commented:
Behat test results file URL: http://openy-dev.ffwua.com/build484/profiles/openy/build/reports/behat

alexschedrov reviewed (approved):
Approved.
